### PR TITLE
Add user validation for age

### PR DESCRIPTION
**Pull request by Vlad6501**

### CODE REVIEW

File: src/main/java/com/kpi/project/service/UserService.java [hunk 1]
```diff
@@ -64,7 +64,7 @@ public User loadUserByUsername(String login) throws UsernameNotFoundException {
     public UserDto saveUser(UserDto userDto) {
         final String userPassword = userDto.getPassword();
         userValidator.validatePassword(userPassword, userDto.getMatchingPassword());
-        userValidator.validateUser(userDto.getEmail(), userDto.getUsername());
+        userValidator.validateUser(userDto.getEmail(), userDto.getUsername(), userDto.getAge());
```
Comment (snahornyi) [inline src/main/java/com/kpi/project/service/UserService.java:67]: too many args, pass userDto

File: src/main/java/com/kpi/project/validate/UserValidator.java [hunk 2]
```diff
@@ -61,13 +62,16 @@ public void validatePassword(String password, String matchingPassword) {
...
-    public void validateUser(String userEmail, String userName) {
+    public void validateUser(String userEmail, String userName, LocalDate age) {
         if (StringUtils.isBlank(userEmail)) {
             throw new ValidatorException("Email should be present");
         }
         if (StringUtils.isBlank(userName)) {
             throw new ValidatorException("Username should be present");
         }
+        if (age.isAfter(LocalDate.now().minusYears(16))) {
+            throw new ValidatorException("insufficient age");
```
Comment (snahornyi) [inline src/main/java/com/kpi/project/validate/UserValidator.java:73]: Need better error massage

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 3]
```diff
@@ -51,15 +52,15 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
     public void validateUserShouldThrowExceptionIfEmailIsNotPresent() {
         // expected
         assertThatExceptionOfType(ValidatorException.class)
-                .isThrownBy(() -> testingInstance.validateUser("", "username"))
+                .isThrownBy(() -> testingInstance.validateUser("", "username", LocalDate.of(2000, 1, 1)))
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:55]: just pass null

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 4]
```diff
@@ -51,15 +52,15 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
...
                 .withMessage("Email should be present");
     }
 
     @Test
     public void validateUserShouldThrowExceptionIfUserNameIsNotPresent() {
         // expected
         assertThatExceptionOfType(ValidatorException.class)
-                .isThrownBy(() -> testingInstance.validateUser("email@mail.com", ""))
+                .isThrownBy(() -> testingInstance.validateUser("email@mail.com", "", LocalDate.of(2000, 1, 1)))
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:63]: pass null

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 5]
```diff
@@ -72,7 +73,7 @@ public void validateUserShouldThrowExceptionIfEmailAlreadyExists() {
 
         // expected
         assertThatExceptionOfType(ValidatorException.class)
-                .isThrownBy(() -> testingInstance.validateUser("email@mail.com", "username"))
+                .isThrownBy(() -> testingInstance.validateUser("email@mail.com", "username", LocalDate.of(2000, 1, 1)))
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:76]: with updates about userDto, than use just givenUser

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 6]
```diff
@@ -146,6 +147,15 @@ public void validateUserExistenceShouldThrowExceptionUserIsNotExist() {
                 .withMessage("User with id : 25, not exists");
     }
 
+    @Test
+    public void validateUserAgeShouldThrowExceptionIfUserAgeIsUnderSixteen() {
+        //expected
+        assertThatExceptionOfType(ValidatorException.class)
+                .isThrownBy(() -> testingInstance.validateUser("some@gmail.com", "user", LocalDate.of(2010, 1, 1)))
+                .withMessage("insufficient age");
+
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:156]: redundant

**Review by snahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/kpi/project/model/dto/UserDto.java [hunk 7]
```diff
@@ -13,6 +15,9 @@
 
     String username;
 
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd/MM/yyyy")
+    LocalDate age;
```
Comment (snahornyi) [inline src/main/java/com/kpi/project/model/dto/UserDto.java:19]: u use date params for date of birth, please change variable name

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 8]
```diff
@@ -155,4 +175,15 @@ private static User givenUser(Function<User.UserBuilder, User.UserBuilder> userC
...
+    private static UserDto givenUserDto(Function<UserDto.UserDtoBuilder, UserDto.UserDtoBuilder> userDtoCustomizer) {
+        return userDtoCustomizer.apply(UserDto.builder()
+                .id(1L)
+                .email("mail@mail.com")
+                .username("username")
+                .age(LocalDate.of(2000, 1, 1))
+                .password("password")
+                .matchingPassword("password"))
+                .build();
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:187]: Do we really need this field, or we can just create simple users

**Review by snahornyi — CHANGES_REQUESTED** [review]

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 9]
```diff
@@ -146,6 +156,16 @@ public void validateUserExistenceShouldThrowExceptionUserIsNotExist() {
                 .withMessage("User with id : 25, not exists");
     }
 
+    @Test
+    public void validateUserAgeShouldThrowExceptionIfUserAgeIsUnderSixteen() {
```
Comment (olegLevochkin) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:160]: Add test if date is good

**Review by olegLevochkin — CHANGES_REQUESTED** [review]

File: src/main/java/com/kpi/project/model/User.java [hunk 10]
```diff
@@ -46,6 +47,9 @@
     @Column(name = "USERNAME", unique = true, nullable = false)
     String username;
 
+    @Column(name = "DAYOFBIRTH", nullable = false)
```
Comment (snahornyi) [inline src/main/java/com/kpi/project/model/User.java:50]: DATE_OF_BIRTH

File: src/main/java/com/kpi/project/model/User.java [hunk 11]
```diff
@@ -46,6 +47,9 @@
...
+    LocalDate dayOfBirth;
```
Comment (snahornyi) [inline src/main/java/com/kpi/project/model/User.java:51]: Rename with updates from previous comment

File: src/main/java/com/kpi/project/model/dto/UserDto.java [hunk 12]
```diff
@@ -14,6 +16,9 @@
 
     String username;
 
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd/MM/yyyy")
+    LocalDate dayOfBirth;
```
Comment (snahornyi) [inline src/main/java/com/kpi/project/model/dto/UserDto.java:20]: rename

File: src/main/java/com/kpi/project/validate/UserValidator.java [hunk 13]
```diff
@@ -67,21 +69,24 @@ public void validatePassword(String password, String matchingPassword) {
...
+        final User userByEmail = userRepository.findByEmail(userDto.getEmail());
+        final User userByUsername = userRepository.findByUsername(userDto.getUsername());
         if (Objects.nonNull(userByEmail)) {
             throw new ValidatorException("Email already exists");
         }
         if (Objects.nonNull(userByUsername)) {
             throw new ValidatorException("Username already exists");
         }
+        if (userDto.getDayOfBirth().isAfter(LocalDate.now().minusYears(16))) {
+            throw new ValidatorException("User must be over sixteen years old");
```
Comment (snahornyi) [inline src/main/java/com/kpi/project/validate/UserValidator.java:89]: Change error message

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 14]
```diff
@@ -49,43 +51,67 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
 
     @Test
     public void validateUserShouldThrowExceptionIfEmailIsNotPresent() {
+        // given
+        final UserDto userDto = UserDto.builder()
+                .username("user")
+                .email(null)
+                .dayOfBirth(LocalDate.of(2010, 1, 1))
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:58]: Do we really need all this data?

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 15]
```diff
@@ -49,43 +51,67 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
...
                 .withMessage("Email should be present");
     }
 
     @Test
     public void validateUserShouldThrowExceptionIfUserNameIsNotPresent() {
+        // given
+        final UserDto userDto = UserDto.builder()
+                .username(null)
+                .email("same@same.com")
+                .dayOfBirth(LocalDate.of(2010, 1, 1))
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:73]: same

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 16]
```diff
@@ -49,43 +51,67 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
...
     }
 
     @Test
     public void validateUserShouldThrowExceptionIfEmailAlreadyExists() {
         // given
         final User givenUser = givenUser(userBuilder -> userBuilder.email("email@mail.com"));
+        final UserDto userDto = UserDto.builder()
+                .username("user")
+                .email("email@mail.com")
+                .dayOfBirth(LocalDate.of(2010, 1, 1))
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:89]: same

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 17]
```diff
@@ -49,43 +51,67 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
...
         assertThatExceptionOfType(ValidatorException.class)
-                .isThrownBy(() -> testingInstance.validateUser("", "username"))
+                .isThrownBy(() -> testingInstance.validateUser(userDto))
                 .withMessage("Email should be present");
     }
 
     @Test
     public void validateUserShouldThrowExceptionIfUserNameIsNotPresent() {
+        // given
+        final UserDto userDto = UserDto.builder()
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:70]: givenUserDto

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 18]
```diff
@@ -49,43 +51,67 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
 
     @Test
     public void validateUserShouldThrowExceptionIfEmailIsNotPresent() {
+        // given
+        final UserDto userDto = UserDto.builder()
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:55]: given UserDto

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 19]
```diff
@@ -49,43 +51,67 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
...
     }
 
     @Test
     public void validateUserShouldThrowExceptionIfUserNameAlreadyExists() {
         // given
         final User givenUser = givenUser(userBuilder -> userBuilder.username("existingUserName"));
+        final UserDto userDto = UserDto.builder()
+                .username("existingUserName")
+                .email("same@same.com")
+                .dayOfBirth(LocalDate.of(2010, 1, 1))
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:107]: redundant data

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 20]
```diff
@@ -154,6 +180,20 @@ public void validateFriendToAddShouldThrowExceptionUserIsNotExist() {
                 .withMessage("User with username: username, not exists");
     }
 
+    @Test
+    public void validateUserAgeShouldThrowExceptionIfUserAgeIsOverSixteen() {
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:184]: rename

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 21]
```diff
@@ -154,6 +180,20 @@ public void validateFriendToAddShouldThrowExceptionUserIsNotExist() {
...
+        //given
+        final UserDto userDto = UserDto.builder()
+                .username("existingUserName")
+                .email("same@same.com")
+                .dayOfBirth(LocalDate.now().minusYears(16).plusDays(1))
+                .build();
+        //expected
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:191]: \n

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 22]
```diff
@@ -49,43 +51,67 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
...
-                .isThrownBy(() -> testingInstance.validateUser("email@mail.com", "username"))
+                .isThrownBy(() -> testingInstance.validateUser(userDto))
                 .withMessage("Email already exists");
     }
 
     @Test
     public void validateUserShouldThrowExceptionIfUserNameAlreadyExists() {
         // given
         final User givenUser = givenUser(userBuilder -> userBuilder.username("existingUserName"));
+        final UserDto userDto = UserDto.builder()
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:104]: givenUserDto

**Review by snahornyi — CHANGES_REQUESTED** [review]
Add test with all correct user args

File: src/main/java/com/kpi/project/validate/UserValidator.java [hunk 23]
```diff
@@ -67,21 +69,24 @@ public void validatePassword(String password, String matchingPassword) {
...
+        final User userByEmail = userRepository.findByEmail(userDto.getEmail());
+        final User userByUsername = userRepository.findByUsername(userDto.getUsername());
         if (Objects.nonNull(userByEmail)) {
             throw new ValidatorException("Email already exists");
         }
         if (Objects.nonNull(userByUsername)) {
             throw new ValidatorException("Username already exists");
         }
+        if (!userDto.getDateOfBirth().isBefore(LocalDate.now().minusYears(16))) {
+            throw new ValidatorException("To register, the user must be over sixteen years old");
```
Comment (snahornyi) [inline src/main/java/com/kpi/project/validate/UserValidator.java:89]: Age restriction of sixteen years

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 24]
```diff
@@ -49,43 +52,62 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
 
     @Test
     public void validateUserShouldThrowExceptionIfEmailIsNotPresent() {
+        // given
+        final UserDto givenUserDto = UserDto.builder()
+                .email(null)
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:57]: default is null already

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 25]
```diff
@@ -49,43 +52,62 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
...
-                .isThrownBy(() -> testingInstance.validateUser("", "username"))
+                .isThrownBy(() -> testingInstance.validateUser(givenUserDto))
                 .withMessage("Email should be present");
     }
 
     @Test
     public void validateUserShouldThrowExceptionIfUserNameIsNotPresent() {
+        // given
+        final UserDto givenUserDto = UserDto.builder()
+                .username(null)
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:70]: same

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 26]
```diff
@@ -49,43 +52,62 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
...
+                .isThrownBy(() -> testingInstance.validateUser(givenUserDto))
                 .withMessage("Username should be present");
     }
 
     @Test
     public void validateUserShouldThrowExceptionIfEmailAlreadyExists() {
         // given
         final User givenUser = givenUser(userBuilder -> userBuilder.email("email@mail.com"));
+        final UserDto givenUserDto = UserDto.builder()
+                .username("user")
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:85]: username

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 27]
```diff
@@ -154,6 +176,30 @@ public void validateFriendToAddShouldThrowExceptionUserIsNotExist() {
...
+                .withMessage("To register, the user must be over sixteen years old");
+    }
+
+    @Test
+    public void validateUserAgeShouldThrowExceptionIfUserAgeIsOlderSixteen() {
+        //given
+        final UserDto givenUserDto = UserDto.builder()
+                .dateOfBirth(LocalDate.now().minusYears(16).minusDays(1))
+                .build();
+        //expected
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:199]: \n

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 28]
```diff
@@ -154,6 +176,30 @@ public void validateFriendToAddShouldThrowExceptionUserIsNotExist() {
...
 
+    @Test
+    public void validateUserAgeShouldThrowExceptionIfUserAgeIsUnderSixteen() {
+        //given
+        final UserDto givenUserDto = UserDto.builder()
+                .username("existingUserName")
+                .email("same@same.com")
+                .dateOfBirth(LocalDate.now().minusYears(16))
+                .build();
+        //expected
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:187]: \n

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 29]
```diff
@@ -154,6 +176,30 @@ public void validateFriendToAddShouldThrowExceptionUserIsNotExist() {
...
+        assertThatExceptionOfType(ValidatorException.class)
+                .isThrownBy(() -> testingInstance.validateUser(givenUserDto))
+                .withMessage("To register, the user must be over sixteen years old");
+    }
+
+    @Test
+    public void validateUserAgeShouldThrowExceptionIfUserAgeIsOlderSixteen() {
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:194]: rename

Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:194]: refactor this test

**Review by snahornyi — CHANGES_REQUESTED** [review]

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 30]
```diff
@@ -49,43 +56,60 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
 
     @Test
     public void validateUserShouldThrowExceptionIfEmailIsNotPresent() {
+        // given
+        final UserDto givenUserDto = UserDto.builder()
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:60]: One line

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 31]
```diff
@@ -49,43 +56,60 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
...
-                .isThrownBy(() -> testingInstance.validateUser("", "username"))
+                .isThrownBy(() -> testingInstance.validateUser(givenUserDto))
                 .withMessage("Email should be present");
     }
 
     @Test
     public void validateUserShouldThrowExceptionIfUserNameIsNotPresent() {
+        // given
+        final UserDto givenUserDto = UserDto.builder()
+                .email("same@same.com")
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:73]: Replace with better words

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 32]
```diff
@@ -49,43 +56,60 @@ public void validateUserShouldThrowExceptionIfPasswordHasIncorrectLength() {
...
                 .withMessage("Email already exists");
     }
 
     @Test
     public void validateUserShouldThrowExceptionIfUserNameAlreadyExists() {
         // given
         final User givenUser = givenUser(userBuilder -> userBuilder.username("existingUserName"));
+        final UserDto givenUserDto = UserDto.builder()
+                .username("existingUserName")
+                .email("same@same.com")
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:105]: same

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 33]
```diff
@@ -154,6 +178,34 @@ public void validateFriendToAddShouldThrowExceptionUserIsNotExist() {
                 .withMessage("User with username: username, not exists");
     }
 
+    @Test
+    public void validateUserAgeShouldThrowExceptionIfUserAgeIsUnderSixteen() {
+        //given
+        final UserDto givenUserDto = UserDto.builder()
+                .username("existingUserName")
+                .email("same@same.com")
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:186]: same

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java (hunk 32 above)
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:105]: same

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 34]
```diff
@@ -154,6 +178,34 @@ public void validateFriendToAddShouldThrowExceptionUserIsNotExist() {
...
+                .isThrownBy(() -> testingInstance.validateUser(givenUserDto))
+                .withMessage("Age restriction of sixteen years");
+    }
+
+    @Test
+    public void validateUserAgeShouldUserAgeIsOlderSixteen() {
+        //given
+        final UserDto givenUserDto = UserDto.builder()
+                .username("userName")
+                .email("same@same.com")
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:201]: same

Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:201]: same

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java (hunk 33 above)
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:186]: same

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 35]
```diff
@@ -154,6 +178,34 @@ public void validateFriendToAddShouldThrowExceptionUserIsNotExist() {
...
+        assertThatExceptionOfType(ValidatorException.class)
+                .isThrownBy(() -> testingInstance.validateUser(givenUserDto))
+                .withMessage("Age restriction of sixteen years");
+    }
+
+    @Test
+    public void validateUserAgeShouldUserAgeIsOlderSixteen() {
+        //given
+        final UserDto givenUserDto = UserDto.builder()
+                .username("userName")
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:200]: username

**Review by snahornyi — CHANGES_REQUESTED** [review]

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 36]
```diff
@@ -154,6 +177,34 @@ public void validateFriendToAddShouldThrowExceptionUserIsNotExist() {
...
+                .build();
+
+        //expected
+        assertThatExceptionOfType(ValidatorException.class)
+                .isThrownBy(() -> testingInstance.validateUser(givenUserDto))
+                .withMessage("Age restriction of sixteen years");
+    }
+
+    @Test
+    public void validateUserAgeShouldUserAgeIsOlderSixteen() {
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:196]: rename

**Review by snahornyi — CHANGES_REQUESTED** [review]

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 37]
```diff
@@ -154,6 +177,34 @@ public void validateFriendToAddShouldThrowExceptionUserIsNotExist() {
...
+                .build();
+
+        //expected
+        assertThatExceptionOfType(ValidatorException.class)
+                .isThrownBy(() -> testingInstance.validateUser(givenUserDto))
+                .withMessage("Age restriction of sixteen years");
+    }
+
+    @Test
+    public void userAgeShouldBeOverSixteen() {
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:196]: rename

**Review by snahornyi — CHANGES_REQUESTED** [review]

File: src/test/java/com/kpi/project/validator/UserValidatorTest.java [hunk 38]
```diff
@@ -154,6 +177,34 @@ public void validateFriendToAddShouldThrowExceptionUserIsNotExist() {
...
+                .build();
+
+        //expected
+        assertThatExceptionOfType(ValidatorException.class)
+                .isThrownBy(() -> testingInstance.validateUser(givenUserDto))
+                .withMessage("Age restriction of sixteen years");
+    }
+
+    @Test
+    public void validateUserShouldNotThrowExceptionIfUserAgeOverSixteen() {
```
Comment (snahornyi) [inline src/test/java/com/kpi/project/validator/UserValidatorTest.java:196]: rename

**Review by snahornyi — CHANGES_REQUESTED** [review]

**Review by snahornyi — APPROVED** [review]

**Review by olegLevochkin — APPROVED** [review]

**Pull request merged**